app/src/main/java/com/aarevalo/tasky/agenda/domain/AgendaItem.kt

```diff
@@ -10,5 +10,6 @@ data class AgendaItem(
     val fromDate: LocalDate = LocalDate.now(),
     val description: String = "Event description",
     val title: String = "New Event",
-    val details: AgendaItemDetails = AgendaItemDetails.Event()
+    val details: AgendaItemDetails = AgendaItemDetails.Event(),
+    val type: AgendaItemType = AgendaItemType.EVENT
```

Comment:
Doesn't the `details` field already contain the type? You could check `if(item.details is AgendaItemDetails.Event)` so that type info seems duplicated/redundant